powersimdata/data_access/launcher.py

```diff
@@ -0,0 +1,207 @@
+import posixpath
+import sys
+
+import requests
+
+from powersimdata.utility import server_setup
+
+
+def _check_threads(threads):
+    """Validate threads argument
+
+    :param int threads: the number of threads to be used
+    :raises TypeError: if threads is not an int
+    :raises ValueError: if threads is not a positive value
+    """
+    if threads:
+        if not isinstance(threads, int):
+            raise TypeError("threads must be an int")
+        if threads < 1:
+            raise ValueError("threads must be a positive value")
+
+
+def _check_solver(solver):
+    """Validate solver argument
+
+    :param str solver: the solver used for the optimization
+    :raises ValueError: if invalid solver provided
+    """
+    solvers = ("gurobi", "glpk")
+    if solver is not None and solver.lower() not in solvers:
```

Comment:
Should we also check `solver` is a str similarly as we did in `_check_threads`?

Reply:
I'm not as worried about that here, since it will still fail due to not matching one of the predefined solvers or not having a `.lower()` method.

Reply:
Yeah, I was thinking about the error message.